dbms/src/Storages/StorageDistributed.cpp

```diff
@@ -481,57 +529,45 @@ bool StorageDistributed::hasColumn(const String & column_name) const
     return virtual_columns.count(column_name) || getColumns().hasPhysical(column_name);
 }
 
-void StorageDistributed::createDirectoryMonitors()
+void StorageDistributed::createDirectoryMonitors(const std::string & disk)
 {
-    if (path.empty())
-        return;
-
+    const std::string path(disk + relative_data_path);
     Poco::File{path}.createDirectories();
 
     std::filesystem::directory_iterator begin(path);
     std::filesystem::directory_iterator end;
     for (auto it = begin; it != end; ++it)
         if (std::filesystem::is_directory(*it))
-            requireDirectoryMonitor(it->path().filename().string());
+            requireDirectoryMonitor(disk, it->path().filename().string());
 }
 
 
-void StorageDistributed::requireDirectoryMonitor(const std::string & name)
+void StorageDistributed::requireDirectoryMonitor(const std::string & disk, const std::string & name)
 {
-    std::lock_guard lock(cluster_nodes_mutex);
-    cluster_nodes_data[name].requireDirectoryMonitor(name, *this, monitors_blocker);
-}
+    const std::string path(disk + relative_data_path + name);
+    LOG_DEBUG(log, "Creating directory monitor for " << path);
```

Comment:
This will contain user:pass, not sure that this is a good idea

Reply:
Yes, better to remove.